ros/processor/inventory_events_consumer.py

```diff
@@ -182,6 +182,16 @@ def _extract_spec_file_data(self, tar, file):
         return extracted_file.read().decode('utf-8')
 
     def _calculate_performance_score(self, performance_record):
+        MAX_IOPS_CAPACITY = 0
         memory_score = (float(performance_record['mem.util.used']) / float(performance_record['mem.physmem'])) * 100
-        performance_score = {'memory_score': int(memory_score)}
+        # considering 16000 as the max iops capacity for volumes in AWS/Azure
+        with app.app_context():
+            cloud_query = (db.session.query(System.cloud_provider).filter(System.id == PerformanceProfile.system_id))
```

Comment:
If the system report is coming for the first time, then this query will return None, because until this point host is not saved in the database. 
Instead of DB query lets pass host object in the function like - `self._calculate_performance_score(performance_record, host)` then you can get the cloud_provider by `host['system_profile']['cloud_provider']`